### ISSUE TITLE
dotnet pack fails with VSTS feeds requiring authentication, while nuget pack works fine (Preview 8)

### ISSUE THREAD

**Issue by IanKemp** [dotnet/core#3231]
Contents of `%appdata%\NuGet\NuGet.Config` (`MyOrgFeed` was added via VS 2017):

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
    <add key="MyOrgFeed" value="https://my-organisation.pkgs.visualstudio.com/_packaging/my-feed-name/nuget/v3/index.json" />
    <add key="Microsoft Visual Studio Offline Packages" value="C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\" />
  </packageSources>
</configuration>
```

`nuget pack` succeeds (note that I am not using nuget.exe directly, but [NuGet.CommandLine 5.2.0](https://chocolatey.org/packages/NuGet.CommandLine)):

```
C:\my-solution-directory> nuget pack my-project\my-project.csproj
Attempting to build package from 'my-project.csproj'.
MSBuild auto-detection: using msbuild version '15.9.21.664' from 'C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\MSBuild\15.0\bin'.
Packing files from 'C:\my-solution-directory\my-project\bin\Debug\netstandard2.0'.
Successfully created package 'C:\my-solution-directory\my-project.1.0.0.nupkg'.
```

but `dotnet pack` fails:

```
C:\my-solution-directory> dotnet pack my-project\my-project.csproj
Microsoft (R) Build Engine version 16.3.0-preview-19377-01+dd8019d9e for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 73.61 ms for C:\my-solution-directory\my-project\my-project.csproj.
C:\Program Files\dotnet\sdk\3.0.100-preview8-013656\NuGet.targets(123,5): error : Unable to load the service index for source https://my-organisation.pkgs.visualstudio.com/_packaging/my-feed-name/nuget/v3/index.json. [C:\my-solution-directory\my-solution-directory.sln]
C:\Program Files\dotnet\sdk\3.0.100-preview8-013656\NuGet.targets(123,5): error :   Response status code does not indicate success: 401 (Unauthorized). [[C:\my-solution-directory\my-solution-directory.sln]]
```

I then went and created a Personal Access Token for Azure DevOps with the only granted permission being "Package: Read", and added it to my `NuGet.config`:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
    <add key="MyOrgFeed" value="https://my-organisation.pkgs.visualstudio.com/_packaging/my-feed-name/nuget/v3/index.json" />
    <add key="Microsoft Visual Studio Offline Packages" value="C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\" />
  </packageSources>
  <packageSourceCredentials>
    <MyOrgFeed>
        <add key="Username" value="example@example.com" />
        <add key="ClearTextPassword" value="azure_personal_access_token" />
    </MyOrgFeed>
  </packageSourceCredentials>
</configuration>
```

Now `dotnet pack` works:

```
C:\my-solution-directory> dotnet pack my-project\my-project.csproj
Microsoft (R) Build Engine version 16.3.0-preview-19377-01+dd8019d9e for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 42.59 ms for C:\Work\EV8\Common\Common\Common.csproj.
  You are using a preview version of .NET Core. See: https://aka.ms/dotnet-core-preview
  Common -> C:\my-solution-directory\my-project\bin\Debug\netstandard2.0\EV8.Common.dll
  Successfully created package 'C:\my-solution-directory\my-project\bin\Debug\my-project.1.0.0.nupkg'.
  Successfully created package 'C:\my-solution-directory\my-project\bin\Debug\my-project.1.0.0.snupkg'.
```

Even more bizarrely, after then removing the PAT from `NuGet.config`, `dotnet pack` **continues to work**!

**Comment by scalablecory** [comment]
@livarcocc @nguerrera @rrelyea can you please look into this?

**Comment by livarcocc** [comment]
This is a NuGet issue. It should not be assigned to my team. @rrelyea's team is probably the best team to take a first look at this.

**Comment by rrelyea** [comment]
If this is still an issue with released 3.0.100, please open an issue in Nuget/home/issues

**Comment by scalablecory** [comment]
Thanks. @IanKemp please open an issue with NuGet if this is still happening with the non-preview 3.0 release.
